### PR TITLE
Generic Assay dropdown: don't show NA when name or description missing (use id instead)

### PR DESCRIPTION
**Pull request by dippindots**
Fix cBioPortal/cbioportal#9041

Describe changes proposed in this pull request:
- Fix Generic Assay options show NA when name or description missing
- Add related unit tests

### CODE REVIEW

File: src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.spec.ts [hunk 1]
```diff
@@ -88,6 +88,61 @@ describe('GenericAssayCommonUtils', () => {
             );
         });
 
+        it('Shows entity_stable_id and description when name is missing in properties', () => {
```
Comment (alisman) [inline src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.spec.ts:91]: these tests seem to assert something different from what the name of the PR implies.  just curious how it works.

Comment (dippindots) [reply src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.spec.ts:91]: @alisman Maybe rename title to `Generic Assay options should not show NA when name or description missing from genericEntityMetaProperties`

Comment (dippindots) [reply src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.spec.ts:91]: This test is for `makeGenericAssayOption` function. These three tests will assert if `NAME` and `DESCRIPTION` property missing from `genericEntityMetaProperties`, will the label still correct.

File: src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts [hunk 2]
```diff
@@ -195,11 +195,13 @@ export function makeGenericAssayOption(
     // indentical to the entity_stable_id.
     const name = getGenericAssayMetaPropertyOrDefault(
         meta,
-        COMMON_GENERIC_ASSAY_PROPERTY.NAME
+        COMMON_GENERIC_ASSAY_PROPERTY.NAME,
+        meta.stableId
     );
```
Comment (dippindots) [inline src/shared/lib/GenericAssayUtils/GenericAssayCommonUtils.ts:200]: The reason why it was showing `NA`:
getGenericAssayMetaPropertyOrDefault will return `NA` when an genericAssayMetaProperty is missing.
```
export function getGenericAssayMetaPropertyOrDefault(
    meta: GenericAssayMeta,
    property: string,
    defaultValue: string = NOT_APPLICABLE_VALUE
): string {
    if (property in meta.genericEntityMetaProperties) {
        return (meta.genericEntityMetaProperties as {
            [property: string]: string;
        })[property];
    } else {
        return defaultValue;
    }
}
```

**Review by alisman — APPROVED** [review]

**Pull request merged**